raiden/tests/integration/api/test_restapi.py

```diff
@@ -1209,3 +1209,92 @@ def test_api_deposit_limit(
     response = response.json()
     assert response['errors'] == 'The deposit of 10001 is bigger than the current limit of 10000'
+
+
+@pytest.mark.parametrize('number_of_nodes', [3])
+def test_channel_history_events(api_backend, raiden_network, token_addresses):
+    _, app1, app2 = raiden_network
+    amount = 200
+    identifier = 42
+    token_address = token_addresses[0]
+
+    target1_address = app1.raiden.address
+    target2_address = app2.raiden.address
+
+    api_server, _ = api_backend
+    our_address = api_server.rest_api.raiden_api.address
+
+    transfer1 = {
+        'initiator_address': to_checksum_address(our_address),
+        'target_address': to_checksum_address(target1_address),
+        'token_address': to_checksum_address(token_address),
+        'amount': amount,
+        'identifier': identifier,
+    }
+
+    # sending tokens to target 1
+    request = grequests.post(
+        api_url_for(
+            api_backend,
+            'transfertotargetresource',
+            token_address=to_checksum_address(token_address),
+            target_address=to_checksum_address(target1_address),
+        ),
+        json={'amount': amount, 'identifier': identifier},
+    )
+    response = request.send().response
+    assert_proper_response(response)
+
+    response = response.json()
+    assert response == transfer1
+
+    # sending some tokens to target 2
+    amount -= 10
+    transfer2 = {
+        'initiator_address': to_checksum_address(our_address),
+        'target_address': to_checksum_address(target2_address),
+        'token_address': to_checksum_address(token_address),
+        'amount': amount,
+        'identifier': identifier,
+    }
+    request = grequests.post(
+        api_url_for(
+            api_backend,
+            'transfertotargetresource',
+            token_address=to_checksum_address(token_address),
+            target_address=to_checksum_address(target2_address),
+        ),
+        json={'amount': amount, 'identifier': identifier},
+    )
+
+    response = request.send().response
+    assert_proper_response(response)
+
+    response = response.json()
+    assert response == transfer2
+
+    # test endpoint without partner
+    request = grequests.get(
+        api_url_for(
+            api_backend,
+            'tokenchannelhistoryresource',
+            partner_address=None,
+            token_address=token_address,
+        ),
+    )
+    response = request.send().response
+    assert_proper_response(response, status_code=HTTPStatus.OK)
+    assert len(response.json()) > 1
+
+    # test endpoint with partner
+    request = grequests.get(
+        api_url_for(
+            api_backend,
+            'channelhistoryresource',
+            partner_address=target1_address,
+            token_address=token_address,
+        ),
+    )
+    response = request.send().response
+    assert_proper_response(response, status_code=HTTPStatus.OK)
+    assert len(response.json()) == 1
```

Comment:
This test is for the event history, and most of the code is testing other behavior.

Please expand this section and make sure the right event is returned with the correct fields.